server/src/main/java/com/networknt/server/Server.java

```diff
@@ -173,8 +175,13 @@ static public void start() {
                 logger.error(errMessage);
                 throw new RuntimeException(errMessage);
             }          
-            for (int i = serverConfig.minPort; i < serverConfig.maxPort; i++) {
-                boolean b = bind(gracefulShutdownHandler, i);
+            while(true) {
+                bindCounter++;
+                if(bindCounter > (serverConfig.maxPort - serverConfig.minPort)) {
+                    break;
+                }
+                int randomPort = ThreadLocalRandom.current().nextInt(serverConfig.minPort, serverConfig.maxPort + 1);
```

Comment:
After research on TreadLoadRandom, if serverConfig.minPort >= serverConfig.maxPort + 1, an IllegalArgumentException would be throw. Should we catch and handle this exception?

Reply:
I think it is a good idea. We shouldn't allow the user to start the server in this case. I will make a change and test it to ensure the server stops. Thanks.

Reply:
There is a check before that. 

```
            if (serverConfig.minPort > serverConfig.maxPort) {
                String errMessage = "No ports available to bind to - the minPort is larger than the maxPort in server.yml";
                System.out.println(errMessage);
                logger.error(errMessage);
                throw new RuntimeException(errMessage);
            }          

```

Reply:
Oh, sorry, I didn't notice that